src/main/java/com/powsybl/caseserver/service/S3CaseService.java

```diff
@@ -0,0 +1,558 @@
+            }
+            // after this line, need to cleanup the dir
+        } catch (IOException e) {
+            throw CaseException.createTempDirectory(caseUuid, e);
+        }
+        try {
+            tempCasePath = tempdirPath.resolve(filename);
+            try {
+                contentInitializer.accept(tempCasePath);
+            } catch (CaseException e) {
+                throw CaseException.initTempFile(caseUuid, e);
+            } catch (Throwable ex) {
+                throw CaseException.initTempFile(caseUuid, ex);
+            }
+            // after this line, need to cleanup the file
+            try {
+                try {
+                    return f.apply(tempCasePath);
+                } catch (CaseException e) {
+                    throw CaseException.createFileNotImportable(tempdirPath);
+                } catch (Throwable t) {
+                    throw CaseException.processTempFile(caseUuid, t);
+                }
+            } finally {
+                try {
+                    Files.delete(tempCasePath);
+                } catch (IOException e) {
+                    LOGGER.error("Error cleaning up temporary case file", e);
+                }
+            }
+        } finally {
+            try {
+                if (Files.exists(tempdirPath)) {
+                    FileUtils.deleteDirectory(tempdirPath.toFile());
+                }
+            } catch (IOException e) {
+                LOGGER.error("Error cleaning up temporary case directory: " + tempdirPath, e);
+            }
+        }
+    }
+
+    // downloads from s3 and cleanup
+    public <R, T extends Throwable> R withS3DownloadedTempPath(UUID caseUuid, FailableFunction<Path, R, T> f) {
+        return withS3DownloadedTempPath(caseUuid, null, f);
+    }
+
+    public <R, T extends Throwable> R withS3DownloadedTempPath(UUID caseUuid, String caseFileKeyFromCaller, FailableFunction<Path, R, T> f) {
+        String caseFileKey = Objects.requireNonNullElse(caseFileKeyFromCaller, getCaseFileObjectKey(caseUuid));
+        String filename = parseFilenameFromKey(caseFileKey);
+        return withTempCopy(caseUuid, filename, path ->
+                        s3Client.getObject(GetObjectRequest.builder().bucket(bucketName).key(caseFileKey).build(), path),
+                f);
+    }
+
+    @Override
+    public String getFormat(UUID caseUuid) {
+        String format = getCaseMetadata(caseUuid).get(FORMAT_HEADER_KEY);
+        if (format == null) {
+            return withS3DownloadedTempPath(caseUuid, this::getFormat);
+        }
+        return format;
+    }
+
+    public String getCompressionFormat(UUID caseUuid) {
+        return caseMetadataRepository.findById(caseUuid).orElseThrow().getCompressionFormat();
+    }
+
+    public String getOriginalFilename(UUID caseUuid) {
+        return caseMetadataRepository.findById(caseUuid).orElseThrow().getOriginalFilename();
+    }
+
+    public Map<String, String> getCaseMetadata(UUID caseUuid) {
```

Comment:
to remove, we don't want to list the files to get the metadata so we don't need an API getMetadata(Uuid caseUuid)

Reply:
done